app/models/item.rb

```diff
@@ -32,10 +32,30 @@ def no_orders?
   end
 
   def self.five_most_popular_items
-    joins(:item_orders).group(:id).select('items.name, sum(quantity) as purchase_amount').order("purchase_amount desc").limit(5)
+    find_by_sql("select items.name, sum(quantity) as purchase_amount from items 
+      inner join item_orders on item_orders.item_id = items.id 
+      inner join orders on orders.id = item_orders.order_id 
+      inner join users on users.id = orders.user_id 
+      where \"users\".\"enabled?\" = 't' 
+      group by items.name 
+      order by purchase_amount desc 
+      limit 5;"
+    )
+    # includes(item_orders: [orders: :users]).group(:id).where('user.enabled? = true').select('item.name, sum(quantity) as purchase_amount').order('purchase_amount desc').limit(5)
```

Comment:
@CoopTang Are the comments for us to reference during our evaluation or were they left in by mistake?

Reply:
They were left by mistake while I was trying to get it to work.

Reply:
I took the liberty of deleting the comments.